gis/gccview/gcc_layer_refresh.py

```diff
@@ -0,0 +1,662 @@
+    """
+    This function calls to the GCCview rest API and inserts the outputs to the output table in the postgres database.
+
+    Parameters
+    ----------
+    mapserver : int
+        The name of the mapserver that host the desire layer
+
+    layer_id : int
+        The id of desired layer
+    
+    schema_name : string
+        The schema in which the table will be inserted into
+    
+    is_audited: Boolean
+        Whether we want to have the table be audited (true) or be partitioned (false)
+    
+    cred: Airflow PostgresHook
+        Contains credentials to enable a connection to a database
+    """
+    successful_task_run = True
+
+    con = cred.get_conn()
+    mapserver = mapserver_name(mapserver_n)
+    output_table = get_tablename(mapserver, layer_id)
+    #--------------------------------
+    if is_audited:
+        primary_key = pk_dict.get(output_table)
+    #--------------------------------
+    keep_adding = True
+    counter = 0
+    
+    while keep_adding == True:
+        
+        if counter == 0:
+            return_json = get_data(mapserver, layer_id)
+            if is_audited:
+                (insert_column, excluded_column) = create_audited_table(output_table, return_json, schema_name, primary_key, con)
+            else:
+                (insert_column, output_table_with_date) = create_partitioned_table(output_table, return_json, schema_name, con)
+            
+            features = return_json['features']
+            record_max=(len(features))
+            max_number = record_max
+            
+            if is_audited:
+                insert_audited_data(output_table, insert_column, return_json, schema_name, con)
+            else:
+                insert_partitioned_data(output_table_with_date, insert_column, return_json, schema_name, con)
+            
+            counter += 1
+            keep_adding = find_limit(return_json)
+            if keep_adding == False:
+                LOGGER.info('All records from [mapserver: %s, layerID: %d] have been inserted into %s', mapserver, layer_id, output_table)
+        else:
+            return_json = get_data(mapserver, layer_id, max_number = max_number, record_max = record_max)
+            if is_audited:
+                insert_audited_data(output_table, insert_column, return_json, schema_name, con)
+            else:
+                insert_partitioned_data(output_table_with_date, insert_column, return_json, schema_name, con)
+            
+            counter += 1
+            keep_adding = find_limit(return_json)
+            if keep_adding == True:
+                max_number = max_number + record_max
+            else:
+                LOGGER.info('All records from [mapserver: %s, layerID: %d] have been inserted into %s', mapserver, layer_id, output_table)
+    
+    if is_audited:
+        successful_task_run = update_table(output_table, insert_column, excluded_column, primary_key, schema_name, con)
+
+    return successful_task_run
```

Comment:
Sorry, but it turns out that just returning True/False isn't enough to mark a task as `FAILED`. The correct way to do it is to raise `AirflowFailException`. So instead of `return successful_task_run`, it could be something like
```
from airflow.exceptions import AirflowFailException
if not successful_task_run:
    raise AirflowFailException
```